Source/Rampage/ViewController.swift

```diff
@@ -26,68 +26,128 @@ private func loadTextures() -> Textures {
 }
 
 class ViewController: UIViewController {
-    private let imageView = UIImageView()
-    private let panGesture = UIPanGestureRecognizer()
+    
+    private let contentView: ContentView = ContentView()
+    
     private let textures = loadTextures()
     private var world = World(map: loadMap())
     private var lastFrameTime = CACurrentMediaTime()
-
+    
+    private var imageView: UIImageView {
+        return self.contentView.imageView
+    }
+    
+    override func loadView() {
+        self.view = contentView
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        setUpImageView()
-
+        
         let displayLink = CADisplayLink(target: self, selector: #selector(update))
         displayLink.add(to: .main, forMode: .common)
-
-        view.addGestureRecognizer(panGesture)
+        
     }
-
-    private var inputVector: Vector {
-        switch panGesture.state {
-        case .began, .changed:
-            let translation = panGesture.translation(in: view)
-            var vector = Vector(x: Double(translation.x), y: Double(translation.y))
-            vector /= max(joystickRadius, vector.length)
-            panGesture.setTranslation(CGPoint(
-                x: vector.x * joystickRadius,
-                y: vector.y * joystickRadius
-            ), in: view)
-            return vector
-        default:
-            return Vector(x: 0, y: 0)
-        }
-    }
-
+    
     @objc func update(_ displayLink: CADisplayLink) {
         let timeStep = min(maximumTimeStep, displayLink.timestamp - lastFrameTime)
-        let inputVector = self.inputVector
-        let rotation = inputVector.x * world.player.turningSpeed * worldTimeStep
+        let leftInputVector = self.contentView.leftJoystickInputVector
+        let rightInputVector = self.contentView.rightJoystickInputVector
+        
+        let rotation = rightInputVector.x * world.player.turningSpeed * worldTimeStep
         let input = Input(
-            speed: -inputVector.y,
+            speed: leftInputVector.orthogonal,
```

Comment:
If you change this to just `Vector(x: leftInputVector.x, y: -leftInputVector.y)` then you can swap `input.speed.x` and `input.speed.y` in the `World.update()` code above, which I think would be more intuitive.